src/gmp.c

```diff
@@ -18527,6 +18577,144 @@ handle_modify_config (gmp_parser_t *gmp_parser, GError **error)
 
 extern char client_address[];
 
+/**
+ * @brief Handle create_report_data->results_* for gmp_xml_handle_end_element
+ *
+ * Uses data:
+ * create_report_data->result_description
+ * create_report_data->result_host
+ * create_report_data->result_hostname
+ * create_report_data->result_nvt_oid
+ * create_report_data->result_port
+ * create_report_data->result_qod
+ * create_report_data->result_qod_type
+ * create_report_data->result_scan_nvt_version
+ * create_report_data->result_severity
+ * create_report_data->result_threat
+ * create_report_data->result_detection_name
+ * create_report_data->result_detection_product
+ * create_report_data->result_detection_source_name
+ * create_report_data->result_detection_source_oid
+ * create_report_data->result_detection_location
+ * create_report_data->result_detection
+ *
+ * to create a create_report_data->result and add it into
+ * create_report_data->results
+ *
+ */
+static void
+gmp_xml_handle_result ()
+{
+  create_report_result_t *result;
+
+  assert (create_report_data->results);
+
+  if (create_report_data->result_scan_nvt_version == NULL)
+    create_report_data->result_scan_nvt_version = strdup ("");
+
+  if (create_report_data->result_severity == NULL)
+    {
+      if (create_report_data->result_threat == NULL)
+        create_report_data->result_severity = strdup ("");
+      else if (strcasecmp (create_report_data->result_threat, "High") == 0)
+        create_report_data->result_severity = strdup ("10.0");
+      else if (strcasecmp (create_report_data->result_threat, "Medium") == 0)
+        create_report_data->result_severity = strdup ("5.0");
+      else if (strcasecmp (create_report_data->result_threat, "Low") == 0)
+        create_report_data->result_severity = strdup ("2.0");
+      else if (strcasecmp (create_report_data->result_threat, "Log") == 0)
+        create_report_data->result_severity = strdup ("0.0");
+      else if (strcasecmp (create_report_data->result_threat, "False Positive")
+               == 0)
+        create_report_data->result_severity = strdup ("-1.0");
+      else
+        create_report_data->result_severity = strdup ("");
+    }
```

Comment:
Is it common practice to not use `{}` for one line conditions in gvmd? I remember this is veeery error-prone ...  Remembering the SSL Bug by apple https://blog.codecentric.de/en/2014/02/curly-braces/

Reply:
Personally I would prefer always using braces even for one line conditions. In the past the gvmd developers avoided the braces in their coding style recommendations.

Reply:
Although I usually also prefer braces, in this `else if` chain construct there isn't that much benefit since it would break compilation with a second line except in the else construct but a
```
if (create_report_data->result_threat == NULL)
{
 ...
}
else
{
  if (...)
  {

  }
 ...
```
would be much harder to read.

Reply:
Huh? So

```C
if (create_report_data->result_threat == NULL)
{
 ...
}
else if (...)
{
  ...
}
 ...
```

wouldn't work?!

Reply:
could be a compromise, yes. 
I still think that for if else chain it is not as critical as suggested within the article though.

Reply:
Honestly I have never ever seen that structure for `if else`
```
else
{
  if (...)
  {

  }
```

Reply:
The preferred way to use curly braces for me would be
```c
if (create_report_data->result_threat == NULL)
  {
    ...
  }
else if (...)
  {
    ...
  }
...
```
In a case like this where there are only simple assignments, I'd also be fine with not using curly braces.

Reply:
You are free to choose whatever you think works best for you. I wont dictate anything ;-) Just get an agreement.

Reply:
I've seen that construct, mostly in projects with overly strict auto-format-rules though.

Since I don't mind either I've pushed an updated version with braces.

@timopollmeier would be ok for you if we run clang-format with the rules across the project? Since I'm used to just run that, but in gmp.c it would have changed even more :).

Reply:
I don't care at all, too.
Just wanted to mention the error-prone-ness of the one-line-conditions ... :)